lib/drivers/bayerContourNext.js

```diff
@@ -0,0 +1,510 @@
+    debug('attempting probe of Bayer Contour Next');
+  };
+
+  var bcnPacketHandler = function (buffer) {
+    // we need to get the REPORT_BYTES here for the first communication
+    // time if they are not set
+    var discardCount = 3;
+    if (discardCount) {
+      buffer.discard(discardCount);
+    }
+
+    if (buffer.len() < 6) { // all complete packets must be at least this long
+      return false;       // not enough there yet
+    }
+
+    // there's enough there to try, anyway
+    var packet = extractPacket(buffer.bytes());
+    if (packet.packet_len !== 0) {
+      // remove the now-processed packet
+      buffer.discard(HID_PACKET_SIZE + 1);  // 1 is for packet_length
+    }
+
+    if (packet.valid) {
+      return packet;
+    } else {
+      return null;
+    }
+  };
+
+  var buildPacket = function (command, cmdlength) {
+    var datalen = cmdlength + 4; // We need just 5 bytes for now
+    var buf = new ArrayBuffer(datalen);
+    var bytes = new Uint8Array(buf);
+    var ctr = struct.pack(bytes, 0, 'bbb', REPORT_BYTES.reportID,
+                          REPORT_BYTES.hostID, REPORT_BYTES.deviceID,
+                          STX, command);
+    if (cmdlength) {
+      ctr += struct.pack(bytes, 3, 'bb', cmdlength, command);
+    }
+    return buf;
+  };
+
+  var buildReadCmd = function(recnum) {
+    return {
+        packet: buildPacket(ASCII_CONTROL.ACK, 1),
+        parser: function(result){
+          var tostr = _.map(result.bytes,
+                            function(e){
+                              return String.fromCharCode(e);
+                            }).join('');
+          result.payload = tostr;
+          return tostr;
+        }
+    };
+  };
+
+  var buildAckPacket = function() {
+    return buildPacket(ASCII_CONTROL.ACK, 1);
+  };
+
+  var buildHeaderPacket = function() {
+    var cmd = 0x3a;  // just anything 'x'
+    return buildPacket(cmd, 1);
+  };
+
+  var extractPacket = function (bytes) {
+    var packet = {
+      bytes: bytes,
+      valid: false,
+      packet_len: 0,
+      payload: null
+    };
+
+    var plen = bytes.length;
+    var packet_len = struct.extractByte(bytes, 0);
+
+    if (packet_len > plen) {
+      return packet;
+    }
+
+    //discard the length byte from the begining
+    var tmpbuff = new ArrayBuffer(packet_len);
+    struct.copyBytes(tmpbuff, 0, bytes, packet_len, 1);
+    packet.bytes = tmpbuff;
+
+    // we now have enough length for a complete packet, so calc the CRC
+    packet.packet_len = packet_len;
+    packet.valid = true;
+
+    return packet;
+  };
+
+
+  var buildHeaderCmd = function() {
+    return {
+      packet: buildAckPacket(),
+      parser: function (result) {
+        var tostr = _.map(result.bytes,
+                          function(e){
+                            return String.fromCharCode(e);
+                          }).join('');
+        result.payload = tostr;
+        return tostr;
+      },
+    };
+  };
+
+  var parseHeader = function (s){
```

Comment:
Please put an example of the header in a comment. This split/shift/pop stuff is dizzying.